### ISSUE TITLE
Workspace is restarted when creating git-credentials secret for the first time

### ISSUE THREAD

**Issue by RomanNikitenko** [eclipse-che/che#22522]
### Describe the bug

Websocket is closed when I create a new git-credentials secret.
I see the same behaviour when I remove git-credentials secret. 

### Che version

next (development version)

### Steps to reproduce

1. Create a workspace
2. Go to the openshift console
3. Create a new git-credentials secret.
4. Track browser logs for the VS Code editor.
5. Wait for the `Cannot reconnect. Please reload the window` dialog.

### Expected behavior

Websocket connection should not be closed when we create/delete git-credentials secret. 

### Runtime

I'm able to reproduce it permanently on the instance from the `Cluster-Bot` with `openshift-v4.13.11` version.
It contains `7.75.0-SNAPSHOT` version of Che

### Screenshots

https://github.com/eclipse/che/assets/5676062/ccf26091-c480-47fd-b122-4881703c01ec

**Comment by tolusha** [comment]
Is it workspace is being restarted at that very moment?

**Comment by RomanNikitenko** [comment]
yes, I just checked 2 times it and 2 times the workspace was restarted.

**Comment by l0rd** [comment]
Can you please check on what version of the DWO it is reproducible?

**Comment by RomanNikitenko** [comment]
I tested few last versions of the `devworkspace-controller`, the result is: I can reproduce the problem for all tested versions.

**Comment by l0rd** [comment]
Opened this issue to discuss UX improvement when the workspace is restarted.

**Comment by RomanNikitenko** [comment]
I remember that I was **not** able to reproduce the problem on the dogfooding instance.
So, I'm curious 
- is it really a bug that affects some version 
or 
- it's configurable

**Comment by amisevsk** [comment]
On the DWO side, since the git credentials secret is mounted to the workspace, creating/removing a secret (when there is only one secret) results in an update to the workspace deployment, which leads to the pod being restarted. However, updating an existing secret should not result in the workspace restarting.

We could work around this by always mounting a (potentially empty) `merged-git-credentials` secret that is only updated with new credentials secrets. This would mean the workspace isn't restarted when the first credentials secret is created/deleted, but I would be worried about applying this functionality to all workspaces automatically (e.g. does this make sense for web terminals). Maybe this functionality could be scoped to a configuration setting in DWO that the Che Operator would always set?

**Comment by l0rd** [comment]
> We could work around this by always mounting a (potentially empty) merged-git-credentials secret that is only updated with new credentials secrets. This would mean the workspace isn't restarted when the first credentials secret is created/deleted, but I would be worried about applying this functionality to all workspaces automatically (e.g. does this make sense for web terminals). Maybe this functionality could be scoped to a configuration setting in DWO that the Che Operator would always set?

I would prefer if Che was responsible of always mounting the secret. I don't remember why we need DWO to handle "merged-git-credential" though.

**Comment by amisevsk** [comment]
DWO handles merged git credentials because the credentials file (that backs the credentials store for git) has to be a single file on disk -- DWO takes all git-credential secrets and merges them into `merged-git-credentials` and mounts that.

**Comment by l0rd** [comment]
Ok, so one way to address that would be that Che creates an empty secret:

```yaml
kind: Secret
apiVersion: v1
metadata:
  name: git-credentials-secret
  labels:
    controller.devfile.io/git-credential: 'true'
    controller.devfile.io/watch-secret: 'true'
type: Opaque
data: {}
```

That's confusing so I don't think it's a good idea. 

And in my opinion this is a rare use case (git hasn't been setup) and it's not worth making any change Che / DW side.

I would rather invest in showing a message to the user if `git` isn't setup in the workspace so that it can configure and restart the workspace early on.

**Comment by l0rd** [comment]
@RomanNikitenko @amisevsk I have updated the title of this issue as, if I understand correctly, it happens only when the `git-credential` secret doesn't exist.
